counterpartylib/lib/messages/versions/enhanced_send.py

```diff
@@ -0,0 +1,202 @@
+
+    unpacked = {
+      'asset': asset,
+      'quantity': quantity,
+      'address': full_address,
+      'memo': memo_bytes,
+    }
+    return unpacked
+
+def validate (db, source, destination, asset, quantity, memo_bytes, block_index):
+    problems = []
+
+    if asset == config.BTC: problems.append('cannot send {}'.format(config.BTC))
+
+    if not isinstance(quantity, int):
+        problems.append('quantity must be in satoshis')
+        return problems
+
+    if quantity < 0:
+        problems.append('negative quantity')
+
+    # if quantity == 0:
+    #     problems.append('zero quantity')
+
+    # For SQLite3
+    if quantity > config.MAX_INT:
+        problems.append('integer overflow')
+
+    # destination is always required
+    if not destination:
+        problems.append('destination is required')
+
+    # check memo
+    if memo_bytes is not None and len(memo_bytes) > MAX_MEMO_LENGTH:
+      problems.append('memo is too long')
+
+    return problems
+
+def compose (db, source, destination, asset, quantity, memo, memo_is_hex):
+    cursor = db.cursor()
+
+    # Just send BTC?
+    if asset == config.BTC:
+        return (source, [(destination, quantity)], None)
+
+    # resolve subassets
+    asset = util.resolve_subasset_longname(db, asset)
+
+    #quantity must be in int satoshi (not float, string, etc)
+    if not isinstance(quantity, int):
+        raise exceptions.ComposeError('quantity must be an int (in satoshi)')
+
+    # Only for outgoing (incoming will overburn).
+    balances = list(cursor.execute('''SELECT * FROM balances WHERE (address = ? AND asset = ?)''', (source, asset)))
+    if not balances or balances[0]['quantity'] < quantity:
+        raise exceptions.ComposeError('insufficient funds')
+
+    # convert memo to memo_bytes based on memo_is_hex setting
+    if memo is None:
+        memo_bytes = b''
+    elif memo_is_hex:
+        memo_bytes = bytes.fromhex(memo)
+    else:
+        memo = memo.encode('utf-8')
+        memo_bytes = struct.pack(">{}s".format(len(memo)), memo)
+
+    block_index = util.CURRENT_BLOCK_INDEX
+
+    problems = validate(db, source, destination, asset, quantity, memo_bytes, block_index)
+    if problems: raise exceptions.ComposeError(problems)
+
+    asset_id = util.get_asset_id(db, asset, block_index)
+
+    short_address_bytes = address.pack(destination)
+
+    data = message_type.pack(ID)
+    data += struct.pack(FORMAT, asset_id, quantity, short_address_bytes)
+    data += memo_bytes
+
+    cursor.close()
+    # return an empty array as the second argument because we don't need to send BTC dust to the recipient
+    return (source, [], data)
+
+def parse (db, tx, message):
+    cursor = db.cursor()
+
+    # Unpack message.
+    try:
+        unpacked = unpack(db, message, tx['block_index'])
+        asset       = unpacked['asset']
+        quantity    = unpacked['quantity']
+        destination = unpacked['address']
+        memo_bytes  = unpacked['memo']
```

Comment:
I dont think this is pythonesque nor do we ever do this aligning of `=` signs.  
could just do `asset, quantity, destination, memo_bytes = unpacked['asset'], unpacked['quantity'], unpacked['address'], unpacked['memo']`